pkg/acs/calacs/lib/mkspt.c

```diff
@@ -191,7 +229,50 @@ int mkNewSpt (char *in_list, char *mtype, char *output) {
         freeHdr (&header);
     }
 
+    /* If necessary, re-open the primary header to update the NEXTEND keyword */
+    if ((extnum != numSPT) || (numSPT != nimgs)) {
+
+        imUpdate = openUpdateImage (out_spt, "", 0, &header);
+        if (hstio_err()) {
+            trlopenerr (out_spt);
+            freeHdr (&header);
+            return (status = OPEN_FAILED);
```

Comment:
_c.f._ previous review [comment](https://github.com/spacetelescope/hstcal/pull/286#discussion_r156191226):

>Also, openUpdateImage can actually set an error and still have opened the file, e.g. for bad dims. In this case, the file needs to be closed before the above return. This being said, responsibility & ownership needs to be considered, i.e. should openUpdateImage close before setting the error and returning?

Reply:
I neglected to make a comment here previously.  IMHO the openUpdateImage function should close the file before it returns an error code to the invoking routine.  This issue falls into the category of cleanup of HSTIO.

Reply:
_c.f._ #289
Looking at the code in more detail, this is the only solution since `BADDIMS` is not persistent and thus there is no means to disambiguate this error from another returned `NULL` ptr.